extensions/gadgets/ScoreBar/ScoreBar.js

```diff
@@ -21,21 +21,35 @@
  */
 
 oppia.directive('oppiaGadgetScoreBar', [
-  'oppiaHtmlEscaper', 'learnerParamsService', function(oppiaHtmlEscaper, learnerParamsService) {
-
+  'oppiaHtmlEscaper', 'learnerParamsService',
+  function(oppiaHtmlEscaper, learnerParamsService) {
     return {
       restrict: 'E',
       templateUrl: 'gadget/ScoreBar',
-      controller: ['$scope', '$attrs', function ($scope, $attrs) {
-
-        $scope.scoreBarLabel = oppiaHtmlEscaper.escapedJsonToObj($attrs.gadgetName);
-        $scope.maxValue = oppiaHtmlEscaper.escapedJsonToObj($attrs.maxValueWithValue);
-        $scope.scoreBarParamName = oppiaHtmlEscaper.escapedJsonToObj($attrs.paramNameWithValue);
+      controller: ['$scope', '$attrs', function($scope, $attrs) {
+        $scope.scoreBarLabel = oppiaHtmlEscaper.escapedJsonToObj(
+          $attrs.gadgetName);
+        $scope.maxValue = oppiaHtmlEscaper.escapedJsonToObj(
+          $attrs.maxValueWithValue);
+        $scope.scoreBarParamName = oppiaHtmlEscaper.escapedJsonToObj(
+          $attrs.paramNameWithValue);
 
         $scope.getScoreValue = function() {
-          return learnerParamsService.getValue($scope.scoreBarParamName);
+          // If a parameter's value exceeds the ScoreBar's maxValue, return
+          // maxValue to keep the label text visible.
+          if (learnerParamsService.getValue(
+              $scope.scoreBarParamName) > $scope.maxValue) {
+            return $scope.maxValue;
+          } else if (learnerParamsService.getValue(
+              $scope.scoreBarParamName) < 0) {
+            // Defense-in-depth: although underflows don't break the current
+            // UI, return a sensible minimum.
+            return 0;
+          } else {
+            return learnerParamsService.getValue($scope.scoreBarParamName);
+          }
```

Comment:
Pushed a new commit that uses clamp. Began work on numberUtilsService, but don't have time to finish it at the moment. I took a note to self to follow up on it or file it as an issue.